src/actions/plugins/targets/smartQuotesMatcher.ts

```diff
@@ -0,0 +1,336 @@
+import { Position } from 'vscode';
+import { TextDocument } from 'vscode';
+import { configuration } from '../../../configuration/configuration';
+
+type Quote = '"' | "'" | '`';
+enum QuoteMatch {
+  Opening,
+  Closing,
+}
+export type WhichQuotes = 'current' | 'next' | 'last';
+type Dir = '>' | '<';
+type SearchAction = {
+  first: Dir;
+  second: Dir;
+  includeCurrent: boolean;
+};
+type QuotesAction = {
+  search: SearchAction | undefined;
+  skipToLeft: number; // for last quotes, how many quotes need to skip while searching
+  skipToRight: number; // for next quotes, how many quotes need to skip while searching
+};
+
+const quoteDirs: Record<string, QuotesAction> = {
```

Comment:
Can you add a short comment explaining what the key is?

Reply:
```suggestion
/**
 * This mapping is used to give a way to identify which action we need to take when operating on a line.
 * The keys here are, in some sense, the number of quotes in the line, in the format of `lcr`, where:
 * `l` means left of the cursor, `c` whether the cursor is on a quote, and `r` is right of the cursor.
 * 
 * It is based on ideas used in `targets.vim`. For each line & cursor position, we count the number of quotes
 * left (#L) and right (#R) of the cursor. Using those numbers and whether the cursor it on a quote, we know 
 * what action to make.
 * 
 * For each entry we have an example of a line & position.
 */
const quoteDirs: Record<string, QuotesAction> = {
```